tajo-catalog/tajo-catalog-client/src/main/java/org/apache/tajo/catalog/AbstractCatalogClient.java

```diff
@@ -446,6 +450,52 @@ public final PartitionDescProto getPartition(final String databaseName, final St
   }
 
   @Override
+  public boolean existPartitions(String databaseName, String tableName) {
+    try {
+      final BlockingInterface stub = getStub();
+      final TableIdentifierProto request = buildTableIdentifier(databaseName, tableName);
+      return isSuccess(stub.existPartitionsByTableName(null, request));
```

Comment:
Even through this API is checking if the table exists, it potentially returns unexpected exception. If you check only its success, it may return a wrong result.

So, you should return true or false depending on only both its success and UNDEFINED_PARTITION_METHODS. You can find the similar code pattern from ``existPartitionMethod`` method in this class.